src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -3979,13 +4020,8 @@ void ComputeIntThermalAbsorpFactors(EnergyPlusData &state)
         for (int SurfNum = firstSurf; SurfNum <= lastSurf; ++SurfNum) {
             int const ConstrNum = Surface(SurfNum).Construction;
             state.dataHeatBal->ITABSF(SurfNum) = state.dataConstruction->Construct(ConstrNum).InsideAbsorpThermal;
-            Real64 HMovInsul = 0.0;                                              // Conductance of movable insulation
-            Real64 AbsInt;                                                       // Solar absorptance of movable insulation
-            if (state.dataConstruction->Construct(ConstrNum).TransDiff <= 0.0) { // Opaque surface
-                if (Surface(SurfNum).MaterialMovInsulInt > 0) EvalInsideMovableInsulation(state, SurfNum, HMovInsul, AbsInt);
-                if (HMovInsul > 0.0)
-                    state.dataHeatBal->ITABSF(SurfNum) =
-                        state.dataMaterial->Material(Surface(SurfNum).MaterialMovInsulInt).AbsorpThermal; // Movable inside insulation present
+            if (state.dataSurface->SurfMovInsulHInt(SurfNum) > 0.0) {
+                state.dataHeatBal->ITABSF(SurfNum) = state.dataMaterial->Material(Surface(SurfNum).MaterialMovInsulInt).AbsorpThermal; // Movable inside insulation present
             }
```

Comment:
Seems that `EvalInsideMovableInsulation` could manage setting `ITABSF`.  Seems this entire loop could be eliminated then. But what about EMS construction changes.

For your todo list: `ComputeIntThermalAbsorpFactors` and `ComputeIntSWAbsorpFactors` appear to be completely recomputed every timestep? Seems these function are ripe for skipping a zone if nothing has changed with window shade status or movable insulation? Not suggesting you try that here.

Reply:
Runtime `SurfAbsThermalExt` is used instead here and initialized with `state.dataSurface->SurfActiveConstruction` at the beginning of simulation or when any EMS override is on.